flows/prefect2_only/deployment_storage.py

```diff
@@ -0,0 +1,87 @@
+import os
+import subprocess
+import sys
+from pathlib import Path
+
+import anyio
+from packaging.version import Version
+
+import prefect
+from prefect.filesystems import LocalFileSystem
+
+# The version oldest version this test runs with
+SUPPORTED_VERSION = "2.6.0"
+
+
+if Version(prefect.__version__) < Version(SUPPORTED_VERSION):
+    raise NotImplementedError()
+
+
+@prefect.flow
+def hello(name: str = "world"):
+    prefect.get_run_logger().info(f"Hello {name}!")
+    return foo() + bar()
+
+
+@prefect.flow
+def foo():
+    return 1
+
+
+@prefect.flow
+async def bar():
+    return 2
+
+
+async def create_flow_run(deployment_id):
+    async with prefect.get_client() as client:
+        return await client.create_flow_run_from_deployment(
+            deployment_id, parameters={"name": "integration tests"}
+        )
+
+
+async def read_flow_run(flow_run_id):
+    async with prefect.get_client() as client:
+        return await client.read_flow_run(flow_run_id)
+
+
+def main():
+    if Version(prefect.__version__) <= Version("2.19.0"):
+        from prefect.deployments import Deployment
```

Comment:
This moves `Deployment` import behind a version check as we won't have `Deployment` after 2.19.0